src/tools/compiletest/src/runtest.rs

```diff
@@ -3517,6 +3517,10 @@ impl<'test> TestCx<'test> {
             .env_remove("MFLAGS")
             .env_remove("CARGO_MAKEFLAGS");
 
+        // In test code we want to be very pedantic about values being silently discarded that are
+        // annotated with `#[must_use]`.
+        cmd.arg("-Dunused_must_use");
```

Comment:
I feel like this should probably respect `deny-warnings = false` in `config.toml`, although not using a `#[must_use]` thing is probably serious enough bug that we should gate compilation for it even during development (vs just having unused variables).

Reply:
I feel like this is different from "normal" code:

- I would not `#[deny(unused_must_use)]` for normal code (e.g. rustc), because there are often cases where it is noisy and it's better checked at CI to avoid blocking local dev.
- But for these rmake.rs tests, they are often short, and not using `#[must_use]` values are almost always bugs and are esepecially bad if that allows the test to silently pass. That is, imo, if the test author *really* wants to not use a `#[must_use]` value, they should probably `let _ = must_use()` or something better.

I'm not opposed to making this respect `deny-warnings = false` in `config.toml`, but if I'm running tests locally I personally *want* them to fail fast due to `-Dunused_must_use` instead of waiting for PR CI.

Reply:
Ok, agreed, let's keep it like this.